docs/content/tutorial/managing-your-own-io.mdx

```diff
@@ -0,0 +1,106 @@
+
+While doing so, you'll take a step back and learn how to use more of Dagster's versatility without using every feature provided to you. Throughout this tutorial, you've learned multiple Dagster concepts that explain how Dagster can manage functionality on your behalf, ex. I/O managers to read and write from storage.
+
+You'll create assets to manage reading and writing from storage on your own. This is useful for multiple purposes, such as:
+
+- Your pipeline manages I/O on its own
+- You want to run a SQL query that creates or updates a table on the database.
+- Adjacent to the above point, your data is too large to fit into memory.
+- You don't want to use I/O managers.
+
+By the end of this section, you'll also have assets that depend on other assets without loading the dependent asset's data into memory.
+
+## Step 1: Creating an asset without an I/O Manager
+
+Many valid situations exist where you would want to avoid using an I/O manager to write an asset back to memory. One is when the asset is already saved in the asset function definition without having to return the data.
+
+In your `assets.py`, create a new asset called `stopwords_zip` by copying and pasting the code below:
+
+```python file=/tutorial/managing_your_own_io/assets.py startafter=start_stopwords_zip endbefore=end_stopwords_zip
+import urllib.request  # Addition: new import at the top of `assets.py`
+
+from dagster import asset
+
+
+@asset
+def stopwords_zip() -> None:
+    urllib.request.urlretrieve(
+        "https://docs.dagster.io/assets/stopwords.zip,data/stopwords.zip",
+    )
+```
+
+In this asset, you’ll download a file from the internet. Dagster stores this file on the machine it is running on. There are two key differences between the assets you’ve written so far and this asset:
+
+- **It doesn't have a return statement.** Because the zip file is downloaded directly onto the machine, there is no need to return data.
+- **It has a `None` annotation for its return type.** This tells Dagster that it shouldn’t use an I/O manager because there is no data to write back.
+
+These are the only two changes required to have an asset not use an I/O manager.
+
+Refresh your code location, observe the new asset, and materialize it as you have with the previous assets. You can confirm that this works by finding the `stopwords.zip` in the `/data` directory made in the **Saving your data** section of the tutorial.
+
+## Step 2: Depend on an asset without loading it
+
+You’ll now unzip the zip file asset you made and extract the CSV from it. This CSV will be a new asset that depends on the zip file asset you downloaded.
+
+By default, an asset will load its dependencies into memory when it materializes. However, there are situations where you would want to avoid doing this. As previously mentioned, if your data is too large to fit into memory, or if you didn’t use an I/O manager for the dependency, you would not want to try loading the dependency.
```

Comment:
I think the main reason that we want to avoid doing this is that we're invoking a library method – `extractall` – that operates on data that's already on disk. So there's no need for us to read/write to disk in an IO manager.